### ISSUE TITLE
Create GH action to check at intervals for the presence of the latest che-server artifacts in Maven Central (and Quay)

### ISSUE THREAD

**Issue by nickboldt** [eclipse-che/che#18832]
### Is your task related to a problem? Please describe.

Recently we had a build process problem where we didn't trap for a maven build failure, and therefore released che-operator even though che-server had not been properly released. The offending script should now be fixed, but the issue remains that we need to have better verifications in place during the release to ensure requirements are met before the next step starts.

### Describe the solution you'd like

- [x] a GH action / process to check at intervals (every 20s) for the presence of latest che-server image. Can use curl + jq. eg., to see if quay.io/eclipse/che-server:7.24.1 exists, use a function like https://github.com/eclipse/che-release/blob/master/cico_common.sh#L41-L70

Until artifacts are found, block starting the che-operator release. We're already doing this \o/

https://github.com/eclipse/che-release/blob/master/cico_release.sh#L624-L628

- [ ] new GH action to check at intervals (every 5 mins?) for the presence of latest **che-server artifacts in Maven Central**. Can use curl + check header for 404 status?

maybe check these 2 artifacts?
* https://search.maven.org/search?q=g:org.eclipse.che%20AND%20a:che-server%20AND%20v:7.24.2
* https://search.maven.org/search?q=g:org.eclipse.che.core%20AND%20a:che-core-api-core%20AND%20v:7.24.2
* (others too?)

eg., can check if pom exists or doesn't exist for a given version:
* https://search.maven.org/remotecontent?filepath=org/eclipse/che/core/che-core-api-core/7.24.2/che-core-api-core-7.24.2.pom (200, exists)
vs.
* https://search.maven.org/remotecontent?filepath=org/eclipse/che/core/che-core-api-core/7.24.6/che-core-api-core-7.24.6.pom (404, not found)

GH action should run in parallel with the main che-release process (launched by API, but not block downstream phases), since:
* it can take 2hrs for artifacts to appear in Maven Central, so shouldn't block release as long as mvn build was successful and che-server image exists in Quay
* we only want it to notify if a problem has occurred (maven artifacts are not needed for Che release, but ARE needed for Hosted Che releases)

Notifications should be:

- [ ] FOR SUCCESS: comment on the current Che Release issue saying that 'Che-server and che-core artifacts are present on Maven Central (with links)'

- [ ] FOR FAILURE / TIMEOUT: post to Eclipse mattermost with news
- [ ] FOR FAILURE / TIMEOUT: email to che-dev@ with news

**Comment by nickboldt** [comment]
No news as of Feb 22. Don't yet have a way to send mail, but can definitely use MM notification

**Comment by nickboldt** [comment]
no need to do this now, as we will be removing maven artifacts as of 7.29. https://github.com/eclipse/che/issues/18941
